ml_inference_offloading/src/main/java/ai/nnstreamer/ml/inference/offloading/MainActivity.kt

```diff
@@ -130,12 +155,14 @@ class MainActivity : ComponentActivity() {
         // Dependency Injection
         (application as App).appComponent.inject(this)
         super.onCreate(savedInstanceState)
-        if (!hasCameraPermission()) {
-            ActivityCompat.requestPermissions(
-                this,
-                arrayOf(Manifest.permission.CAMERA),
-                0
-            )
+
+        val info: PackageInfo =
+            packageManager.getPackageInfo(applicationContext.packageName, PackageManager.GET_PERMISSIONS)
+        val permissions = info.requestedPermissions
+        if (permissions != null) {
+            if (!permissions.all { ContextCompat.checkSelfPermission(this, it) == PackageManager.PERMISSION_GRANTED }) {
+                requestPermissionLauncher.launch(permissions)
+            }
         }
```

Comment:
```info.requestedPermissions``` is mutable so that you can use ```let`` to  ensure checked value is immutable.
```suggestion
        info.requestedPermissions?.let { permissions ->
            // Your Code
        }
```

In addition, ```!permissions.all(//condition)``` looks weird.
```kotlin
permissions.any(ContextCompat.checkSelfPermission(this, it) != PackageManager.PERMISSION_GRANTED)
```

Reply:
Thanks a lot, Updated!